### PR TITLE
feat: Add vision support

### PR DESCRIPTION
**Pull request by TheRamU**
- Gemini Pro Vision supported
- GPT-4 Vision supported
- Fixed not reporting to the user when Gemini request error
- Image features are now globally supported in this application

### CODE REVIEW

**Comment by H0llyW00dzZ** [comment]
LGTM

File: app/store/chat.ts [hunk 1]
```diff
@@ -84,11 +85,20 @@ function createEmptySession(): ChatSession {
 
 function getSummarizeModel(currentModel: string) {
   // if it is using gpt-* models, force to use 3.5 to summarize
-  return currentModel.startsWith("gpt") ? SUMMARIZE_MODEL : currentModel;
+  if (currentModel.startsWith("gpt")) {
+    return SUMMARIZE_MODEL;
+  }
+  if (currentModel.startsWith("gemini-pro")) {
```
Comment (H0llyW00dzZ) [inline app/store/chat.ts:91]: > [!TIP]
> It's more effective to use `Summarize` with Gemini's own models. For example, `gemini-pro` refers to the Gemini Pro model, and `gemini-pro-vision` refers to the Gemini Pro Vision model. These models are more affordable compared to `OpenAI` hahaha and more efficient, as each model has its own specific token limit. For instance, the token input limit for the `gemini-pro-vision` model is around `12288 tokens`.

Proof:

image

**Review by H0llyW00dzZ — CHANGES_REQUESTED** [review]
Request Change `Summarize` for `gemini-pro` and `gemini-pro-vision`

**Comment by H0llyW00dzZ** [comment]
Show off

image

File: app/store/chat.ts (hunk 1 above)
Comment (H0llyW00dzZ) [reply app/store/chat.ts:91]: > Gemini Pro Vision does not enable multiturn chat, which limits its role as a "Summarize Model"

rip, its better model, unlike openai `gpt-4-vision-preview`

**Comment by fengzai6** [comment]
能否在选中谷歌视觉模型的时候，对话不使用多轮聊天，仅上传单论对话，不用每次使用需要清除上下文

**Comment by fengzai6** [comment]
能否在选中谷歌视觉模型的时候，对话不使用多轮聊天，仅上传单次对话，不用每次使用需要清除上下文

**Comment by PengLingJun** [comment]
Why are my responses using the vision-preview api incomplete

image

image

**Pull request merged**